covid19_sfbayarea/data/sonoma.py

```diff
@@ -117,24 +117,83 @@ def transform_cases(cases_tag: element.Tag) -> Dict[str, TimeSeries]:
 
     return { 'cases': cases, 'deaths': deaths }
 
-def transform_transmission(transmission_tag: element.Tag) -> Dict[str, int]:
+def transform_transmission(transmission_tag: element.Tag, total_cases: int, normalize: bool = True) -> Dict[str, int]:
     """
     Takes in a BeautifulSoup tag for the transmissions table and breaks it into
-    a dictionary of type:
-    {'community': -1, 'from_contact': -1, 'travel': -1, 'unknown': -1}
+    a dictionary. Fields are either the original from data source or are normalized
+    into groups consistent with other datasets, by using `normalize=True` (default).
+    Data model:
+      {'community': -1, 'from_contact': -1, 'travel': -1, 'unknown': -1}
     """
     transmissions = {}
     rows = parse_table(transmission_tag)
+
     # turns the transmission categories on the page into the ones we're using
-    transmission_type_conversion = {'Community': 'community', 'Close Contact': 'from_contact', 'Travel': 'travel', 'Under Investigation': 'unknown'}
+    transmission_type_conversion = {
+        'congregate care': 'congregate_care',
+        'health care': 'health_care',
+        'household': 'household',
+        'large gathering': 'gathering_large',
+        'other': 'other',
+        'small gathering': 'gathering_small',
+        'travel': 'travel',
+        'unknown': 'unknown',
+        'workplace': 'workplace'
+    }
+
     assert_equal_sets(transmission_type_conversion.keys(),
-                      (row['Source'] for row in rows),
+                      (row['Exposure Location'].lower() for row in rows),
                       description='Transmission types')
+
     for row in rows:
-        type = row['Source']
-        number = parse_int(row['Cases'])
+        type = row['Exposure Location'].lower()
+        percent_cases = float(row['All Time'].replace('%', '')) / 100
+        case_count = int(percent_cases * total_cases)
         type = transmission_type_conversion[type]
-        transmissions[type] = number
+        transmissions[type] = case_count
+
+    if normalize:
+        # normalize categories into groups consistent with other datasets
+        # those groups are "from_contact", "travel", "unknown" and "community"
+
+        from_contact_categories = [
+            "congregate_care",
+            "household",
+            "workplace",
+            "gathering_small"
+        ]
+        from_contact = sum(
+            [transmissions.get(category, 0) for category in transmissions.keys()
+             if category in from_contact_categories]
+        )
```

Comment:
This could probably be a little more compact and simple:

```suggestion
        from_contact = sum(transmissions[category]
                           for category in from_contact_categories)
```

(You don’t need the `[]` if a generator instead of a list is OK, and if it’s the only argument. It’s also simpler in this case to iterate through jut the categories you care about, and it’s safe to use indexed lookup instead of `.get()` because we already asserted that all these categories have values.)

Same for the repeat of this pattern below.